packages/server-core/src/social/location/location.hooks.ts

```diff
@@ -210,23 +233,24 @@ export default {
     find: [discardQuery('action'), discardQuery('studio'), sortByLocationSetting],
     get: [],
     create: [
-      iff(isProvider('external'), verifyScope('location', 'write')),
       () => schemaHooks.validateData(locationDataValidator),
       schemaHooks.resolveData(locationDataResolver),
+      iff(isProvider('external'), verifyScope('editor', 'write'), verifyProjectPermission(['owner', 'editor'])),
```

Comment:
Can this be made to work in hybrid with either location:write OR project permission with editor:write? Same goes for other methods in this file.

@HexaField what are your thoughts?

Reply:
you cannot verify if user has one scope or the other cause  verifyScope will throw Forbidden the moment one fail not allowing the second check to take place. for that we'll have to modify the hook and that has impact on other hooks.

Reply:
Well there is a checkScope method which returns boolean instead of throw. You can make use of it. Here is the link of code where its used:

https://github.com/EtherealEngine/etherealengine/blob/4bf3b5166a5ad5cf7706eb58d8b43b52441fbf2c/packages/server-core/src/user/user-avatar/user-avatar.hooks.ts#L63

Alternatively you can:

1. Create a new hook which does what verifyScope does, and rather than throwing it returns true false?
2. Update verifyScope hook to have an argument which has a param of throw = true by default. And if you change it to false, then it will return boolean?

Reply:
I just created a new hook in following PR, you can use it in your code it returns true/false instead of throwing an error. Hope it helps?

https://github.com/EtherealEngine/etherealengine/pull/10279

Reply:
Resolved in https://github.com/EtherealEngine/etherealengine/pull/10214/commits/112195766f42fe72364f2acdfaab9e6de1b9e189